src/box/alter.cc

```diff
@@ -5225,6 +5227,8 @@ on_replace_dd_func_index(struct trigger *trigger, void *event)
 			diag_set(ClientError, ER_NO_SUCH_FUNCTION, int2str(fid));
 			return -1;
 		}
+		if (func_access_check(func) != 0)
```

Comment:
Done.

I see the gollowing function check was also redundant, replaced it with a clarifying comment.

Reply:
Investigated this a bit. It turns out the check (along with the following `func_index_check_func`) is used during recovery, so I've returned them.

Reply:
Investigated this a bit. It turns out the check (along with the following func_index_check_func) is used during recovery, so it will fail if the snapshot is broken or inattentively modified manually.